diff --git a/devel.install b/devel.install
index 99a3007..cda2a7a 100644
--- a/devel.install
+++ b/devel.install
@@ -5,6 +5,8 @@
  * Install, update and uninstall functions for the devel module.
  */
 
+use Drupal\Component\Utility\DeprecationHelper;
+use Drupal\Core\Extension\Requirement\RequirementSeverity;
 use Drupal\devel\Plugin\Devel\Dumper\VarDumper;
 
 /**
@@ -34,7 +36,7 @@ function devel_requirements($phase) {
     $requirements['devel'] = [
       'title' => t('Devel module enabled'),
       'description' => t("The Devel module provides access to internal debugging information; therefore it's recommended to disable this module on sites in production."),
-      'severity' => REQUIREMENT_INFO,
+      'severity' => DeprecationHelper::backwardsCompatibleCall(\Drupal::VERSION, '11.2.0', fn() => RequirementSeverity::Info, fn() => REQUIREMENT_INFO),
     ];
   }
 
