diff --git a/src/TagContainerResolver.php b/src/TagContainerResolver.php
index 783852b..41007b6 100644
--- a/src/TagContainerResolver.php
+++ b/src/TagContainerResolver.php
@@ -92,7 +92,7 @@ final class TagContainerResolver {
     if ($request === NULL) {
       return NULL;
     }
-    if (!$this->resolved->contains($request)) {
+    if (!$this->resolved->offsetExists($request)) {
       $storage = $this->entityTypeManager->getStorage('google_tag_container');
       $config_ids = $storage->getQuery()
         // @todo remove after https://github.com/mglaman/phpstan-drupal/issues/479
