diff --git a/metatag.install b/metatag.install
index 8a26785dcc7242af9533ef39d5492b4a238af5bf..31f375d7c0cbc6c48a85eac2128f2c961e159766 100644
--- a/metatag.install
+++ b/metatag.install
@@ -5,6 +5,8 @@
  * Requirements and update functions for the Metatag module.
  */
 
+use Drupal\Component\Utility\DeprecationHelper;
+use Drupal\Core\Extension\Requirement\RequirementSeverity;
 use Drupal\Core\StringTranslation\TranslatableMarkup;
 
 /**
@@ -18,7 +20,7 @@ function metatag_requirements($phase) {
     // mode.
     if (\Drupal::state()->get('system.maintenance_mode')) {
       $requirements['metatag_maintenance_mode'] = [
-        'severity' => REQUIREMENT_WARNING,
+        'severity' => DeprecationHelper::backwardsCompatibleCall(\Drupal::VERSION, '11.2.0', fn() => RequirementSeverity::Warning, fn() => REQUIREMENT_WARNING),
         'title' => 'Metatag',
         'value' => new TranslatableMarkup('Not enabled while in maintenance mode'),
         'description' => new TranslatableMarkup('Please note that while the site is in maintenance mode none of the usual meta tags will be output.'),
@@ -28,7 +30,7 @@ function metatag_requirements($phase) {
     // Recommend the Schema.org Metatag module.
     if (!\Drupal::moduleHandler()->moduleExists('schema_metatag')) {
       $requirements['metatag_schema'] = [
-        'severity' => REQUIREMENT_INFO,
+        'severity' => DeprecationHelper::backwardsCompatibleCall(\Drupal::VERSION, '11.2.0', fn() => RequirementSeverity::Info, fn() => REQUIREMENT_INFO),
         'title' => 'Metatag',
         'value' => new TranslatableMarkup('Schema.org Metatag is recommended'),
         'description' => new TranslatableMarkup('The <a href="@module">Schema.org Metatag</a> module is highly recommended to add <a href="@jsonld">JSON-LD</a> -formatted <a href="@schema">schema.org</a> compatible data structures to the site.', [
@@ -40,7 +42,7 @@ function metatag_requirements($phase) {
     }
     else {
       $requirements['metatag_schema'] = [
-        'severity' => REQUIREMENT_OK,
+        'severity' => DeprecationHelper::backwardsCompatibleCall(\Drupal::VERSION, '11.2.0', fn() => RequirementSeverity::OK, fn() => REQUIREMENT_OK),
         'title' => 'Metatag',
         'value' => new TranslatableMarkup('Schema.org Metatag is installed'),
         'description' => new TranslatableMarkup('The <a href="@module">Schema.org Metatag</a> module is installed.', [
@@ -52,7 +54,7 @@ function metatag_requirements($phase) {
     // Suggest the Token OR module.
     if (!\Drupal::moduleHandler()->moduleExists('token_or')) {
       $requirements['metatag_token_or'] = [
-        'severity' => REQUIREMENT_INFO,
+        'severity' => DeprecationHelper::backwardsCompatibleCall(\Drupal::VERSION, '11.2.0', fn() => RequirementSeverity::Info, fn() => REQUIREMENT_INFO),
         'title' => 'Metatag',
         'value' => new TranslatableMarkup('Token Or'),
         'description' => new TranslatableMarkup('The <a href="@module">Token OR module</a> is suggested for when the basic content tokens are not flexible enough, e.g. to make a meta tag show the contents of one field if it is filled in or another if the first one is empty.', [
