diff --git a/token.install b/token.install
index 2e161db..25bca41 100644
--- a/token.install
+++ b/token.install
@@ -5,6 +5,9 @@
  * Install, update and uninstall functions for the token module.
  */
 
+use Drupal\Component\Utility\DeprecationHelper;
+use Drupal\Core\Extension\Requirement\RequirementSeverity;
+
 /**
  * Implements hook_requirements().
  */
@@ -95,23 +98,23 @@ function token_get_token_problems() {
   $token_problems = [
     'not-array' => [
       'label' => t('Tokens or token types not defined as arrays'),
-      'severity' => REQUIREMENT_ERROR,
+      'severity' => DeprecationHelper::backwardsCompatibleCall(\Drupal::VERSION, '11.2.0', fn() => RequirementSeverity::Error, fn() => REQUIREMENT_ERROR),
     ],
     'missing-info' => [
       'label' => t('Tokens or token types missing name property'),
-      'severity' => REQUIREMENT_WARNING,
+      'severity' => DeprecationHelper::backwardsCompatibleCall(\Drupal::VERSION, '11.2.0', fn() => RequirementSeverity::Warning, fn() => REQUIREMENT_WARNING),
     ],
     'type-no-tokens' => [
       'label' => t('Token types do not have any tokens defined'),
-      'severity' => REQUIREMENT_INFO,
+      'severity' => DeprecationHelper::backwardsCompatibleCall(\Drupal::VERSION, '11.2.0', fn() => RequirementSeverity::Info, fn() => REQUIREMENT_INFO),
     ],
     'tokens-no-type' => [
       'label' => t('Token types are not defined but have tokens'),
-      'severity' => REQUIREMENT_INFO,
+      'severity' => DeprecationHelper::backwardsCompatibleCall(\Drupal::VERSION, '11.2.0', fn() => RequirementSeverity::Info, fn() => REQUIREMENT_INFO),
     ],
     'duplicate' => [
       'label' => t('Token or token types are defined by multiple modules'),
-      'severity' => REQUIREMENT_ERROR,
+      'severity' => DeprecationHelper::backwardsCompatibleCall(\Drupal::VERSION, '11.2.0', fn() => RequirementSeverity::Error, fn() => REQUIREMENT_ERROR),
     ],
   ];
 
