diff --git a/modules/webform_bootstrap/webform_bootstrap.install b/modules/webform_bootstrap/webform_bootstrap.install
index 8dc2308..1c7d138 100644
--- a/modules/webform_bootstrap/webform_bootstrap.install
+++ b/modules/webform_bootstrap/webform_bootstrap.install
@@ -5,6 +5,9 @@
  * Installation information for the 'Webform Bootstrap module.
  */
 
+use Drupal\Component\Utility\DeprecationHelper;
+use Drupal\Core\Extension\Requirement\RequirementSeverity;
+
 /**
  * Implements hook_requirements().
  */
@@ -23,7 +26,7 @@ function webform_bootstrap_requirements() {
       'title' => t('Webform Bootstrap'),
       'value' => t('Bootstrap 3.x missing.'),
       'description' => t('Bootstrap 3.x. must be enabled to install the Webform Bootstrap module.'),
-      'severity' => REQUIREMENT_ERROR,
+      'severity' => DeprecationHelper::backwardsCompatibleCall(\Drupal::VERSION, '11.2.0', fn() => RequirementSeverity::Error, fn() => REQUIREMENT_ERROR),
     ];
   }
   return $requirements;
diff --git a/modules/webform_node/webform_node.install b/modules/webform_node/webform_node.install
index c8a144b..b747477 100644
--- a/modules/webform_node/webform_node.install
+++ b/modules/webform_node/webform_node.install
@@ -5,6 +5,9 @@
  * Install, update and uninstall functions for the webform node module.
  */
 
+use Drupal\Component\Utility\DeprecationHelper;
+use Drupal\Core\Extension\Requirement\RequirementSeverity;
+
 /**
  * Implements hook_requirements().
  */
@@ -20,7 +23,7 @@ function webform_node_requirements($phase) {
         'title' => t('Webform Node'),
         'value' => t('%title content type already exists', ['%title' => $node_type->label()]),
         'description' => t('%title content type already exists, please delete the %title content type before installing the Webform node module.', ['%title' => $node_type->label()]),
-        'severity' => REQUIREMENT_ERROR,
+        'severity' => DeprecationHelper::backwardsCompatibleCall(\Drupal::VERSION, '11.2.0', fn() => RequirementSeverity::Error, fn() => REQUIREMENT_ERROR),
       ];
     }
 
diff --git a/src/Hook/WebformRequirementsHooks.php b/src/Hook/WebformRequirementsHooks.php
index e0c2dff..5616581 100644
--- a/src/Hook/WebformRequirementsHooks.php
+++ b/src/Hook/WebformRequirementsHooks.php
@@ -3,6 +3,8 @@
 namespace Drupal\webform\Hook;
 
 use Drupal\Component\Render\FormattableMarkup;
+use Drupal\Component\Utility\DeprecationHelper;
+use Drupal\Core\Extension\Requirement\RequirementSeverity;
 use Drupal\Core\Config\ConfigFactoryInterface;
 use Drupal\Core\Extension\ModuleExtensionList;
 use Drupal\Core\Extension\ModuleHandlerInterface;
@@ -59,7 +61,7 @@ class WebformRequirementsHooks {
           '@module_list' => implode(', ', $deprecated),
           ':url' => 'https://www.drupal.org/project/webform_deprecated',
         ]),
-        'severity' => REQUIREMENT_ERROR,
+        'severity' => DeprecationHelper::backwardsCompatibleCall(\Drupal::VERSION, '11.2.0', fn() => RequirementSeverity::Error, fn() => REQUIREMENT_ERROR),
       ];
     }
     // Sort all requirements alphabetically.
@@ -95,7 +97,7 @@ class WebformRequirementsHooks {
           '%module_list' => implode(', ', $experimental),
           ':url' => 'https://www.drupal.org/core/experimental',
         ]),
-        'severity' => REQUIREMENT_WARNING,
+        'severity' => DeprecationHelper::backwardsCompatibleCall(\Drupal::VERSION, '11.2.0', fn() => RequirementSeverity::Warning, fn() => REQUIREMENT_WARNING),
       ];
     }
 
@@ -114,7 +116,7 @@ class WebformRequirementsHooks {
       $requirements['webform_email'] = [
         'title' => t('Webform: HTML email support'),
         'value' => ($mail_module_name) ? t('Provided by the @module module.', $t_args) : t('Provided by the @plugin_id mail plugin.', $t_args),
-        'severity' => REQUIREMENT_OK,
+        'severity' => DeprecationHelper::backwardsCompatibleCall(\Drupal::VERSION, '11.2.0', fn() => RequirementSeverity::OK, fn() => REQUIREMENT_OK),
       ];
       if ($mail_plugin_definition) {
         $requirements['webform_email']['description'] = (!empty($mail_plugin_definition['description']))
@@ -126,7 +128,7 @@ class WebformRequirementsHooks {
       $requirements['webform_email'] = [
         'title' => t('Webform: HTML email support'),
         'value' => t('Unable to determine email module and/or provider'),
-        'severity' => REQUIREMENT_ERROR,
+        'severity' => DeprecationHelper::backwardsCompatibleCall(\Drupal::VERSION, '11.2.0', fn() => RequirementSeverity::Error, fn() => REQUIREMENT_ERROR),
       ];
     }
 
@@ -146,7 +148,7 @@ class WebformRequirementsHooks {
         'title' => t('Webform: Private files'),
         'value' => t('Private file system is not set.'),
         'description' => t('This must be changed in <a href="https://www.drupal.org/documentation/modules/file">settings.php</a>. For more information see: <a href="https://www.drupal.org/psa-2016-003">DRUPAL-PSA-2016-003</a>'),
-        'severity' => REQUIREMENT_WARNING,
+        'severity' => DeprecationHelper::backwardsCompatibleCall(\Drupal::VERSION, '11.2.0', fn() => RequirementSeverity::Warning, fn() => REQUIREMENT_WARNING),
       ];
     }
 
@@ -173,7 +175,7 @@ class WebformRequirementsHooks {
           'title' => t('Webform: Clientside validation'),
           'value' => t('Webform Clientside Validation module not installed.'),
           'description' => t('The Webform Clientside Validation module helps support Webform Clientside Validation integration. <a href=":href">Disable Webform Clientside Validation warning</a>', [':href' => Url::fromRoute('webform.config.advanced')->toString()]),
-          'severity' => REQUIREMENT_WARNING,
+          'severity' => DeprecationHelper::backwardsCompatibleCall(\Drupal::VERSION, '11.2.0', fn() => RequirementSeverity::Warning, fn() => REQUIREMENT_WARNING),
         ];
       }
     }
@@ -199,7 +201,7 @@ class WebformRequirementsHooks {
         'title' => t('Webform: jQuery UI Datepicker'),
         'value' => t('Webform jQuery UI Datepicker module not installed.'),
         'description' => t('The Webform jQuery UI Datepicker module is required to support datepickers. Disable this datepicker warning by <a href=":href">removing #datepicker property from all webforms</a>.', $t_args),
-        'severity' => REQUIREMENT_WARNING,
+        'severity' => DeprecationHelper::backwardsCompatibleCall(\Drupal::VERSION, '11.2.0', fn() => RequirementSeverity::Warning, fn() => REQUIREMENT_WARNING),
       ];
     }
 
@@ -283,7 +285,7 @@ class WebformRequirementsHooks {
           'title' => t('Webform: Spam protection'),
           'value' => t('Webform <a href=":href">Spam protection module</a> missing. Please install one of the below modules.', [':href' => 'https://www.drupal.org/node/206787']),
           'description' => $this->renderer->renderInIsolation($available_projects),
-          'severity' => REQUIREMENT_WARNING,
+          'severity' => DeprecationHelper::backwardsCompatibleCall(\Drupal::VERSION, '11.2.0', fn() => RequirementSeverity::Warning, fn() => REQUIREMENT_WARNING),
         ];
       }
     }
diff --git a/src/WebformLibrariesManager.php b/src/WebformLibrariesManager.php
index 7484fc5..1d05b2c 100644
--- a/src/WebformLibrariesManager.php
+++ b/src/WebformLibrariesManager.php
@@ -2,6 +2,8 @@
 
 namespace Drupal\webform;
 
+use Drupal\Component\Utility\DeprecationHelper;
+use Drupal\Core\Extension\Requirement\RequirementSeverity;
 use Drupal\Core\Asset\LibraryDiscoveryInterface;
 use Drupal\Core\Config\ConfigFactoryInterface;
 use Drupal\Core\Extension\ModuleHandlerInterface;
@@ -97,7 +99,7 @@ class WebformLibrariesManager implements WebformLibrariesManagerInterface {
     }
 
     // Track stats.
-    $severity = REQUIREMENT_OK;
+    $severity = DeprecationHelper::backwardsCompatibleCall(\Drupal::VERSION, '11.2.0', fn() => RequirementSeverity::OK, fn() => REQUIREMENT_OK);
     $stats = [
       '@total' => count($libraries),
       '@installed' => 0,
@@ -150,7 +152,7 @@ class WebformLibrariesManager implements WebformLibrariesManagerInterface {
         $stats['@missing']++;
         $title = $this->t('<span class="color-warning"><strong>@title @version</strong> (CDN).</span>', $t_args);
         $description = $this->t('The <a href=":homepage_href">@title</a> library is not installed in <b>@path</b>.', $t_args);
-        $severity = REQUIREMENT_WARNING;
+        $severity = DeprecationHelper::backwardsCompatibleCall(\Drupal::VERSION, '11.2.0', fn() => RequirementSeverity::Warning, fn() => REQUIREMENT_WARNING);
       }
       else {
         // CDN.
@@ -190,7 +192,7 @@ class WebformLibrariesManager implements WebformLibrariesManagerInterface {
 
     // Description.
     $description = [];
-    if (!$cli && $severity === REQUIREMENT_WARNING) {
+    if (!$cli && $severity === DeprecationHelper::backwardsCompatibleCall(\Drupal::VERSION, '11.2.0', fn() => RequirementSeverity::Warning, fn() => REQUIREMENT_WARNING)) {
       $t_args = [':href' => ($this->moduleHandler->moduleExists('help')) ? Url::fromRoute('help.page', ['name' => 'webform'], ['fragment' => 'libraries'])->toString() : 'https://www.drupal.org/docs/8/modules/webform/webform-libraries'];
       $description['download'] = [
         '#markup' => '<hr/>' .
