diff --git a/src/Hook/XmlsitemapViewsHooks.php b/src/Hook/XmlsitemapViewsHooks.php
index 7dfd6e7..88ca679 100644
--- a/src/Hook/XmlsitemapViewsHooks.php
+++ b/src/Hook/XmlsitemapViewsHooks.php
@@ -83,6 +83,9 @@ class XmlsitemapViewsHooks {
   #[Hook('views_data_alter')]
   public function viewsDataAlter(array &$data) {
     foreach (\Drupal::entityTypeManager()->getDefinitions() as $id => $definition) {
+      if (!$definition->getBaseTable()) {
+        continue;
+      }
       if (!xmlsitemap_link_entity_check_enabled($id) && !empty($data[$definition->getBaseTable()])) {
         continue;
       }
